File: PortProject.Api/Application/StorageAreas/DTOs/StorageAreaDto.cs
namespace PortProject.Api.Application.StorageAreas.DTOs;

public class StorageAreaDto
{
    public string Code { get; set; }
    public string Type { get; set; }
    public string Location { get; set; }
    public int Capacity { get; set; }
    public int CurrentOccupancy { get; set; }
}

File: port-spa-app/src/infrastructure/repositories/storageArea/storageArea.dto.ts
// These DTOs (Data Transfer Objects) define the *exact shape*
// of the data we send to and receive from the API.

export interface StorageAreaCreateDto {
    type: string;
    location: string;
    capacity: number;
    currentOccupancy: number;
}

export interface StorageAreaUpdateDto {
    type: string;
    location: string;
    capacity: number;
    currentOccupancy: number;
}

